docs/introduction.md

```diff
@@ -67,83 +98,202 @@ Flow is based upon the following set of rules:
+- Execution Receipt is broadcast to the Verification Nodes and Consensus Nodes
 
-For the sake of load-balancing, redundancy and [Byzantine resilience](#byzantine-fault), the collection nodes are [staked](#staking) equally and randomly positioned into clusters of roughly identical size.
-At the beginning of an epoch, each collection node is randomly assigned to exactly one cluster.
-Each cluster of collection nodes acts as a gateway to Flow, from the external world.
-This clustering mechanism avoids heterogeneous systems where a collection node with better service would be getting all the traffic and end up reducing the decentralization of the whole system as well as starving out other collections.
+#### Collection retrieval
 
-External clients submit their transactions to collection nodes. Upon receiving well-formed transactions, collection nodes introduce them to the rest of their cluster.
-The collection nodes of one cluster batch the received transactions into collections. Only a hash reference to a collection is submitted to the consensus nodes for inclusion in a block.
+Block being executed contains a number of guaranteed collections. Guaranteed collections only includes a hash of the set of transactions, not their full texts.
+Execution Node retrieves the collection text from the appropriate cluster, and reconstructs the collection hash from the transaction text to verify the data consistency.
+Transactions are processed in order.
+
+#### Block Execution
+
+A block is executed when the previous blocks Execution Result is known.
+This previous Execution Result is the starting point for the execution of the block.
+
+Execution Receipt for a block includes Execution Nodes signature, Excution Result and [SPoCKs](#specialized-proof-of-confidential-knowledge).
+Execution Receipt is a signed Execution Result.
+The Execution Receipts can be used to challenge the claims of an execution node when they are shown to be incorrect.
+They are also used to create proofs of the current state of the chain once they are known to be correct.
 
-Each cluster of collection nodes generates their collections one at a time. Before a new collection is started, the current one has to be closed and sent to the consensus nodes for inclusion in a block.
-The collection nodes' consensus protocol determines when to start/end a collection and which transactions to include in the collection. The result of that consensus is called a _guaranteed collection_.
+Execution Result of a block includes:
+- block hash
+- hash reference to the previous execution result
+- one or more chunks
+- commitment to the final execution state of the system after executing the block
 
-##### Verification Nodes
+Execution result has one or more **chunks**.
+**Chunking** is a process of dividing blocks computation into smaller pieces so that individual chunks can be executed and verified in a distributed and parallel manner by many verification nodes.
```

Comment:
```suggestion
**Chunking** is a process of dividing a block's computation into smaller pieces so that individual chunks can be executed and verified in a distributed and parallel manner by many verification nodes.
```